core/io/marshalls.cpp

```diff
@@ -1814,6 +1814,82 @@ Error encode_variant(const Variant &p_variant, uint8_t *r_buffer, int &r_len, bo
 	return OK;
 }
 
+Error to_raw_bytes(const Variant &p_variant, PackedByteArray &out) {
+	switch (p_variant.get_type()) {
+		case Variant::INT: {
+			int32_t v = p_variant;
+			size_t offset = out.size();
+			out.resize(out.size() + sizeof(v));
+			memcpy(out.ptrw() + offset, &v, sizeof(v));
+		} break;
+		case Variant::BOOL: {
+			bool v = p_variant;
+			size_t offset = out.size();
+			out.resize(out.size() + sizeof(v));
+			memcpy(out.ptrw() + offset, &v, sizeof(v));
+		} break;
+		case Variant::FLOAT: {
+#ifdef REAL_T_IS_DOUBLE
+			double v = p_variant;
+			size_t offset = out.size();
+			out.resize(out.size() + sizeof(v));
+			memcpy(out.ptrw() + offset, &v, sizeof(v));
+#else
+			float v = p_variant;
+			size_t offset = out.size();
+			out.resize(out.size() + sizeof(v));
+			memcpy(out.ptrw() + offset, &v, sizeof(v));
+#endif
+		} break;
+		case Variant::VECTOR2: {
+			Vector2 v = p_variant;
+			for (size_t elements = 0; elements < 2; elements++) {
+				to_raw_bytes(v[elements], out);
+			}
+		} break;
+		case Variant::VECTOR3: {
+			Vector3 v = p_variant;
+			for (size_t elements = 0; elements < 3; elements++) {
+				to_raw_bytes(v[elements], out);
+			}
+		} break;
+		case Variant::VECTOR4: {
+			Vector4 v = p_variant;
+			for (size_t elements = 0; elements < 4; elements++) {
+				to_raw_bytes(v[elements], out);
+			}
+		} break;
+		case Variant::VECTOR2I: {
+			Vector2i v = p_variant;
+			for (size_t elements = 0; elements < 2; elements++) {
+				to_raw_bytes(v[elements], out);
+			}
```

Comment:
This is not correct either, the values of `Vector2/3/4i` are all 32 bits, whereas the `int` is always 64 bits, this needs to be handled specifically

Reply:
int and uint in GLSL are 32 bit integers, though you are correct about GDScript int's, they are 64 bits and this have to be adressed.

Reply:
This is fixed with the commit [3266fe5f9da822baba14b2998cf9823405ebdfc1](https://github.com/godotengine/godot/commit/3266fe5f9da822baba14b2998cf9823405ebdfc1), by converting doubles and uint64_t of the GDScript into 32 bit floats and integers.
Documentation updated: [3e7b1bf9417c03c547bb12f802e14d24c2617287](https://github.com/godotengine/godot/commit/3e7b1bf9417c03c547bb12f802e14d24c2617287)